streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/controller/VariableController.java

```diff
@@ -0,0 +1,118 @@
+
+import org.apache.streampark.console.base.domain.RestRequest;
+import org.apache.streampark.console.base.domain.RestResponse;
+import org.apache.streampark.console.base.exception.ApiAlertException;
+import org.apache.streampark.console.core.entity.Application;
+import org.apache.streampark.console.core.service.CommonService;
+import org.apache.streampark.console.system.entity.Variable;
+import org.apache.streampark.console.system.service.VariableService;
+
+import com.baomidou.mybatisplus.core.metadata.IPage;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.shiro.authz.annotation.RequiresPermissions;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotBlank;
+
+import java.util.List;
+
+@Slf4j
+@Validated
+@RestController
+@RequestMapping("variable")
+public class VariableController {
+
+    @Autowired
+    private CommonService commonService;
+
+    @Autowired
+    private VariableService variableService;
+
+    @PostMapping("list")
+    @RequiresPermissions("variable:view")
+    public RestResponse variableList(RestRequest restRequest, Variable variable) {
+        IPage<Variable> variableList = variableService.findVariables(variable, restRequest);
+        return RestResponse.success(variableList);
+    }
+
+    @PostMapping("post")
+    @RequiresPermissions("variable:add")
+    public RestResponse addVariable(@Valid Variable variable) throws Exception {
+        boolean isExists = this.variableService.findByVariableCode(variable.getTeamId(), variable.getVariableCode()) != null;
+        if (isExists) {
+            throw new ApiAlertException("Sorry, the variable code already exists.");
+        }
+        variable.setCreator(commonService.getCurrentUser().getUserId());
+        this.variableService.createVariable(variable);
+        return RestResponse.success();
+    }
+
+    @PutMapping("update")
+    @RequiresPermissions("variable:update")
+    public RestResponse updateVariable(@Valid Variable variable) throws Exception {
+        Variable findVariable = this.variableService.findByVariableCode(variable.getTeamId(), variable.getVariableCode());
+        if (findVariable == null) {
+            throw new ApiAlertException("Sorry, the variable does not exist.");
+        }
+        if (findVariable.getId().longValue() != variable.getId().longValue()) {
+            throw new ApiAlertException("Sorry, the variable id is inconsistent.");
+        }
+        this.variableService.updateVariable(variable);
```

Comment:
This logic is wired, in general, we always findById, then check `findVariable.teamId== variable.teamId && findVariable.getVariableCode() == variable.getVariableCode()`.

Also, I'm not sure these checks should in Controller or Service. @wolfboys Do we have any related specifications before?

Reply:
> This logic is wired, in general, we always findById, then check `findVariable.teamId== variable.teamId && findVariable.getVariableCode() == variable.getVariableCode()`.
> 
> Also, I'm not sure these checks should in Controller or Service. @wolfboys Do we have any related specifications before?

@1996fanrui I will replace these checks with variableService.checkParam(variable)

Reply:
@1996fanrui I will put them in the service.